ci/test/cargo-test/mzcompose.py

```diff
@@ -47,25 +49,91 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
         f"postgres://postgres:postgres@localhost:{c.default_port('postgres')}"
     )
     cockroach_url = f"postgres://root@localhost:{c.default_port('cockroach')}"
-    spawn.runv(
+
+    env = dict(
+        os.environ,
+        ZOOKEEPER_ADDR=f"localhost:{c.default_port('zookeeper')}",
+        KAFKA_ADDRS="localhost:30123",
+        SCHEMA_REGISTRY_URL=f"http://localhost:{c.default_port('schema-registry')}",
+        POSTGRES_URL=postgres_url,
+        COCKROACH_URL=cockroach_url,
+        MZ_SOFT_ASSERTIONS="1",
+        MZ_PERSIST_EXTERNAL_STORAGE_TEST_S3_BUCKET="mtlz-test-persist-1d-lifecycle-delete",
+        MZ_PERSIST_EXTERNAL_STORAGE_TEST_POSTGRES_URL=cockroach_url,
+    )
+
+    coverage = bool(os.getenv("CI_COVERAGE_ENABLED", False))
+
+    if coverage:
```

Comment:
There are 4 `if coverage` tests in this file, some of them straight up code flows and others are ternary operators. Consider refactoring this in a way that would simplify the logic. One option would be to just have a basically separate `workflow_coverage` here, as the non-coverage and coverage flows share just two `spawn.runv(` invocations as far as I can see, and with wildly disjoint arguments for each.